docs/user/tutorial-asgi.rst

```diff
@@ -31,43 +31,41 @@ WSGI tutorial::
       ├── __init__.py
       └── app.py
 
-.. note::
-   Installing `virtualenv <https://docs.python-guide.org/dev/virtualenvs/>`_ is
-   not needed for recent Python 3.x versions. We can simply create a
-   *virtualenv* using the ``venv`` module from the standard library,
-   for instance::
+Installing `virtualenv <https://virtualenv.pypa.io/>`_ is not usually needed
+for recent Python 3.x versions. We can simply create a *virtualenv* using the
+``venv`` module from the standard library::
```

Comment:
Now with the extra clarification added as part of this PR, maybe that `virtualenv` clarification got slightly repetitive indeed.

But maybe it doesn't hurt to stress it twice since Ubuntu and Debian are very popular (at least on the server/container side), and they do package in such a way that it is easy to install Python without the `venv` module. (Offtopic: this, together with different default directory choices, is a source of friction between Debian/Ubuntu and the upstream CPython maintainers IIRC.)

Reply:
At one point debian shipped with a broken wheel package. I was "fun" debugging pep517 problems in sqlalchemy.

Regarding the comment, that's just my opinion, reading it I found a bit repetitive, but I was not aware of the ubuntu/debian drama

Reply:
I don't think there is much drama there _per se_. But from the user perspective it is still annoying that none of the sides is willing to concede, and issues arise due to things such mundane as directories named `dist-packages` vs `site-packages`.

Here's Debian's reasoning: https://wiki.debian.org/Python#Deviations_from_upstream

Reply:
FWIW, I usually opt to install only the bare minimum from the Ubuntu repos, then [bootstrap](https://pip.pypa.io/en/stable/installing/#installing-with-get-pip-py) `pip` globally, and then use that `pip` to install the global packages that I need (which is usually only a handful of basic tools such as `virtualenv`, `virtualenvwrapper` etc).

Reply:
I do basically the same, but instead of using apt I just install conda and use it to install whatever version of python I need. That way I don't have to use wired backports repositories to get up to date versions, that expecially on debian are a bit late to come.

Using conda to manage the python versions and the environments so far has always worker fairly well for me.

Reply:
I agree that the paragraph could be simplified since we do include the note.